include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,815 @@
+                "picked by the openPMD API)"
+            )(
+                (pluginPrefix + ".compression").c_str( ),
+                po::value< std::string >( & compressionMethod )
+                    ->default_value( "" ),
+                "Backend-specific openPMD compression method, e.g., zlib "
+                "(see `adios_config -m` for help)"
+            )
+            (
+                (pluginPrefix + ".memoryLayout").c_str( ),
+                po::value< std::string >( & memoryLayout )
+                    ->default_value( "mirror" ),
+                "Possible values: 'mirror' and 'split'"
+                "Output can be mirrored on all Host+Device pairs or"
+                " uniformly split over all nodes. Split can be used "
+                "when the output array is to big to store the complete "
+                "computed q-space on one device."
+            );
+        }
+
+
+        //! Get plugin name.
+        std::string pluginGetName( ) const override { return pluginName; }
+
+
+        //! Sets Mapping description for the saxs plugin.
+        void setMappingDescription( MappingDesc * cellDescription ) override
+        {
+            this->cellDescription = cellDescription;
+        }
+
+
+        void restart(
+            uint32_t timeStep,
+            const std::string restartDirectory
+        ) override
+        {
+            // TODO: Support for restarting.
+        }
+
+
+        void checkpoint(
+            uint32_t timeStep,
+            const std::string restartDirectory
+        ) override
+        {
+            // TODO: Support for restarting.
+        }
+
+
+private:
+
+
+        //! Prepare the plugin in the simulation initialization phase.
+        void pluginLoad( ) override
+        {
+            if ( !notifyPeriod.empty( ) )
+            {
+                /* Beam has to be initialized later as the domain sizes.
+                 *   The value retrieved by getDomainSize in
+                 *   CoordinateTransform.hpp is still set to (0,0,0) when the
+                 *   Saxs object is initialized.
+                 */
+                probingBeam = pmacc::memory::makeUnique< beam::SaxsProbingBeam
+                    >( );
+                // Set the steps at which the saxs amplitude is calculated.
+                Environment<>::get( ).PluginConnector( ).setNotificationPeriod(
+                    this,
+                    notifyPeriod
+                );
+                // Set the memory layout in use.
+                std::map< std::string, OutputMemoryLayout > layoutMap;
+                layoutMap["mirror"] = OutputMemoryLayout::MirrorOnNodes;
+                layoutMap["split"] = OutputMemoryLayout::SplitOverNodes;
+                outputLayout = layoutMap.at( memoryLayout );
+
+                mpiRank = gc.getGlobalRank( );
+                isMaster = ( mpiRank == 0 );
+
+                // Prepare amplitude buffer:
+                DataSpace< DIM1 > bufferSize;
```

Comment:
There seems to be a bug. `bufferSize` is assigned to in only one of the two following branches, and later used to allocate memory. Taking into account my earlier comments about the `bufferSize_i` member at its declaration, I believe the `bufferSize_i`  should be removed, and in this method `bufferSize` used instead.